enterprise/backend/src/metabase_enterprise/advanced_permissions/models/connection_impersonation.clj

```diff
@@ -60,9 +60,10 @@
 (defn- delete-impersonations-for-group! [{:keys [group-id]} changes]
   (log/debugf "Deleting unneeded Connection Impersonation policies for Group %d. Graph changes: %s" group-id (pr-str changes))
   (doseq [database-id (set (keys changes))]
-    (delete-impersonations-for-group-database!
-     {:group-id group-id, :database-id database-id}
-     (get-in changes [database-id :data :schemas]))))
+    (when-let [data-perm-changes (get-in changes [database-id :data :schemas])]
```

Comment:
If I'm understanding your change, you're just ensuring that `nil` is never passed into the function `delete-impersonations-for-group-database!` right?

I wonder if it makes more sense to capture the nil check in the function directly?

`(when (and (not= :impersonated changes) (not (nil? changes)))` or something like that.

Since it's a private fn anyway, I don't consider this a big deal.

Reply:
I actually had something like that originally and changed it to this. My reasoning is that it's more explicit that we're pulling out `:data :schemas` from the graph, and when that doesn't exist, we skip the connection impersonation update. 

When we're doing a `nil` check inside the function it's not super clear what circumstances might result in a `nil` being there.

Reply:
Ah, that's a really good point. Good thinking!